packages/components/progress/src/less/progress.less

```diff
@@ -115,6 +113,40 @@
     }
   }
 
+  &_size-l {
+    &::before {
+      border-radius: (@oui-progress-height_size-l / 2);
+    }
+
+    &::before,
+    .oui-progress__bar::before {
+      height: @oui-progress-height_size-l;
+    }
+
+    .oui-progress__bar {
+      &::before {
+        border-top-left-radius: (@oui-progress-height_size-l / 2);
+        border-bottom-left-radius: (@oui-progress-height_size-l / 2);
+      }
+
+      & ~ .oui-progress__bar::before {
+        border-top-left-radius: 0;
+        border-bottom-left-radius: 0;
+      }
+
+      &:last-of-type::before {
+        border-top-right-radius: (@oui-progress-height_size-l / 2);
+        border-bottom-right-radius: (@oui-progress-height_size-l / 2);
+      }
+    }
+  }
+
+  &_no-labels {
+    .oui-progress__label {
+      display: none;
```

Comment:
Why not using `ng-if` on the span element in packages/components/progress/src/js/bar/progress-bar.html ?